src/circuits/gates.jl

```diff
@@ -325,36 +335,54 @@ gate(::GateName"RandomUnitary", dims::Tuple = (2,); kwargs...) =
 # qudit gates
 #
 
+function gate(::GateName"Id", dims::Tuple = (2,))
+  g = 1.0
+  for k in 1:length(dims)
+    g = kron(g, Matrix(I,dims[k],dims[k]))
+  end
+  return g
+end
+
+gate(::GateName"Φ", dims::Tuple = (2,); Φ::Number) =
+  exp(im * Φ) * gate("Id", dims)
+
+
 function gate(::GateName"a", dims::Tuple = (2,))
-  @assert length(dims) == 1
-  dim = dims[1]
-  mat = zeros(dim, dim)
-  for k in 1:dim-1
-    mat[k,k+1] = √k
+  #@assert length(dims) == 1
+  #dim = dims[1]
+  #return [(k == j + 1) ? √j : 0.0 for j in 1:dim, k in 1:dim]
+  ignore() do 
```

Comment:
Was Zygote actually trying to differentiate this?

Also, for this case it is probably best to define:
```julia
@non_differentiable gate(::GateName"a", ::Tuple)
```
outside of the function, instead of `ignore()` inside of the function.

Reply:
This was an attempt to debug an issue, and was left there. I'm removing it.

Reply:
This does not appear to be working

Reply:
Which part isn't working?

Reply:
I resolved this.